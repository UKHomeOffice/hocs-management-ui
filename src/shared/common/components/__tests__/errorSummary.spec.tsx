import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from 'enzyme';
import ErrorSummary from '../errorSummary';

const errors = [
    { key: 'Error1', value: 'Error 1' },
    { key: 'Error2', value: 'Error 2' },
    { key: 'Error3', value: 'Error 3' }
];

describe('Form text component', () => {
    it('should render with default props', () => {
        expect(
            render(<ErrorSummary description="Displaying a list of the errors on the page" />)
        ).toMatchSnapshot();
    });
    it('should render with heading when passed', () => {
        expect(
            render(<ErrorSummary description="Displaying a list of the errors on the page" heading="Error summary" />)
        ).toMatchSnapshot();
    });
    it('should render with description when passed', () => {
        expect(
            render(<ErrorSummary description="Displaying a list of the errors on the page" />)
        ).toMatchSnapshot();
    });
    it('should render list of errors when passed', () => {

        expect(
            render(
                < MemoryRouter >
                    <ErrorSummary description="Displaying a list of the errors on the page" errors={errors} />
                </MemoryRouter >
            )
        ).toMatchSnapshot();
    });
});
