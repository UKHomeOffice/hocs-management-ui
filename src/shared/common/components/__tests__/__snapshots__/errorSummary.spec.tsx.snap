// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form text component should render list of errors when passed 1`] = `
<div
  aria-labelledby="error-summary-heading-example-1"
  class="govuk-error-summary"
  role="alert"
  tabindex="-1"
>
  <h2
    class="govuk-error-summary__title"
    id="error-summary-heading-example-1"
  >
    There's a problem
  </h2>
  <div
    class="govuk-error-summary__body"
  >
    <p>
      Displaying a list of the errors on the page
    </p>
    <ul
      class="govuk-list govuk-error-summary__list"
    >
      <li>
        <a
          href="#Error1-error"
        >
          Error 1
        </a>
      </li>
      <li>
        <a
          href="#Error2-error"
        >
          Error 2
        </a>
      </li>
      <li>
        <a
          href="#Error3-error"
        >
          Error 3
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Form text component should render with default props 1`] = `null`;

exports[`Form text component should render with description when passed 1`] = `null`;

exports[`Form text component should render with heading when passed 1`] = `null`;
